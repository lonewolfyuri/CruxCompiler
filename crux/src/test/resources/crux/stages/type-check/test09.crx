// test logical not

func main() : void {
  var i:int;
  var b:bool;

  ::printBool(not i);
  ::printBool(not b);
}
