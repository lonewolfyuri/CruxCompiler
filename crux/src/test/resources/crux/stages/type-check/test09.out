Error type-checking file.
TypeError(7,14)[cannot negate int]
TypeError(7,2)[cannot call func(TypeList(bool)):void using TypeList(ErrorType(cannot negate int))]
